File: src/main/java/com/leijendary/spring/microservicetemplate/model/SampleTable.java
package com.leijendary.spring.microservicetemplate.model;

import com.leijendary.spring.microservicetemplate.model.listener.SampleTableListener;
import lombok.Data;
import lombok.EqualsAndHashCode;
import org.springframework.data.annotation.CreatedBy;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedBy;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EntityListeners;
import java.time.OffsetDateTime;

@Data
@EqualsAndHashCode(callSuper = true)
@Entity
@EntityListeners({ AuditingEntityListener.class, SampleTableListener.class })
public class SampleTable extends LocalizedModel<SampleTable, SampleTableTranslation> {

    @Column(name = "column_1")
    private String column1;

    @Column(name = "column_2")
    private int column2;

    @CreatedDate
    private OffsetDateTime createdDate;

    @CreatedBy
    private String createdBy;

    @LastModifiedDate
    private OffsetDateTime lastModifiedDate;

    @LastModifiedBy
    private String lastModifiedBy;
}

File: src/main/kotlin/com/leijendary/spring/template/model/SampleTable.kt
package com.leijendary.spring.template.model

import com.leijendary.spring.template.core.model.AuditingUUIDModel
import com.leijendary.spring.template.core.model.LocalizedModel
import com.leijendary.spring.template.core.model.SoftDeleteModel
import jakarta.persistence.*
import jakarta.persistence.FetchType.EAGER
import org.hibernate.annotations.Where
import java.math.BigDecimal
import java.math.BigDecimal.ZERO
import java.time.LocalDateTime

@Entity
@Where(clause = "deleted_at is null")
class SampleTable : AuditingUUIDModel(), LocalizedModel<SampleTableTranslation>, SoftDeleteModel {
    @Column(name = "column_1")
    var column1: String = ""

    @Column(name = "column_2")
    var column2 = 0L

    var amount: BigDecimal = ZERO

    @ElementCollection(fetch = EAGER)
    @CollectionTable(name = "sample_table_translation", joinColumns = [JoinColumn(name = "id")])
    override val translations: Set<SampleTableTranslation> = HashSet()

    override var deletedAt: LocalDateTime? = null
    override var deletedBy: String? = null
}
